### PR TITLE
Belly subsystem -> master

### PR DESCRIPTION
**Pull request by NoamRosenberg08**

### CODE REVIEW

File: src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/RotateOutOfDoor.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+public class RotateOutOfDoor extends RotatingBellyCommand{
+
+    private static final double endTimeSeconds = 1;
+    @Override
+    public void execute() {
+        belly.setSpeed(RobotMap.RotatingBelly.PERCENTAGE_ROTATING_SPEED);
+    }
+
+    @Override
+    public boolean isFinished() {
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/RotateOutOfDoor.java:15]: you can remove the time management part from this command and run the command as deadlinecommand with a waitcommad(timeout)

File: src/main/java/edu/greenblitz/tobyDetermined/subsystems/RotatingBelly.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        macroSwitch = new DigitalInput(RobotMap.RotatingBelly.MACRO_SWITCH_PORT);
+        colorSensor = IntakeGameObjectSensor.getInstance();
+        motor = new GBSparkMax(RobotMap.RotatingBelly.MOTOR_ID, CANSparkMaxLowLevel.MotorType.kBrushless);
+    }
+
+    public void setSpeed (double percentageSpeed){
+        motor.set(percentageSpeed);
+    }
+
+    public static RotatingBelly getInstance (){
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/subsystems/RotatingBelly.java:26]: this after constructor

File: src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/FullRotateToPlace.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package edu.greenblitz.tobyDetermined.commands.rotatingBelly;
+
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+
+public class FullRotateToPlace extends SequentialCommandGroup {
+    public FullRotateToPlace (){
+        addCommands(new RotateOutOfDoor());
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/FullRotateToPlace.java:7]: ParallelDeadlineGroup(new RotateOutOfDoor(), new WaitCommand(1));

Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/FullRotateToPlace.java:7]: that's not what it should do. it should rotate backwards for 1 sec, then rotate until macroswitch

File: src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/RotateOutOfDoor.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import edu.greenblitz.tobyDetermined.RobotMap;
+import edu.wpi.first.wpilibj2.command.ParallelDeadlineGroup;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+
+public class RotateOutOfDoor extends ParallelDeadlineGroup {
+
+    public RotateOutOfDoor() {
+        super(
+                new WaitCommand(1),
```
Comment (yonatan-orca) [inline src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/RotateOutOfDoor.java:11]: Isn't it supposed to be the other order? I think what this command is saying is "wait a second until rotateByPower is finished" and it is supposed to by the other way around

Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/tobyDetermined/commands/rotatingBelly/RotateOutOfDoor.java:11]: nope, first is the deadline, then the commands

**Pull request merged**